provider/core/src/error.rs

```diff
@@ -20,6 +20,10 @@ pub enum Error {
     #[displaydoc("Missing resource for request: {0}")]
     MissingResourceOptions(DataRequest),
 
+    /// The data provider supports the key, but the key needs to be supplied with a variant
+    #[displaydoc("Request {0} needs a variant")]
+    MissingVariant(DataRequest),
```

Comment:
Nit: The name here should be `NeedsVariant`.

Note the naming pattern:

- `MissingResourceKey`
- `MissingResourceOptions`

Those are "500 errors": the fault of the data provider.  Whereas

- `NeedsLanguageIdentifier`
- `NeedsTypeInfo`

are "400 errors": the fault of the client calling the data provider.